docs/how-to/configure-text-generation-models.md

```diff
@@ -0,0 +1,160 @@
+catalog:
+  gemma2-2b-cpu:
+    enabled: true
+    engine: OLlama
+    resourceProfile: cpu:2
+    minReplicas: 1 # by default this is 0
+EOF
+```
+
+### Install Text Generation Model using L4 GPU
+
+Enable the Llama 3.1 8B model using the Helm chart:
+
+```bash
+helm upgrade --install --reuse-values kubeai-models kubeai/models -f - <<EOF
+catalog:
+  llama-3.1-8b-instruct-fp8-l4:
+    enabled: true
+    engine: VLLM
+    resourceProfile: nvidia-gpu-l4:1
+    minReplicas: 1 # by default this is 0
+EOF
+```
+
+## Install a Text Generation Model using kubectl
+You can use the Model CRD directly to install a model using `kubectl apply -f model.yaml`.
+
+### Install Text Generation Model using CPU
+
+Apply the following Model CRD to install the Gemma 2 2B model using Ollama on CPU:
+```yaml
+apiVersion: kubeai.org/v1
+kind: Model
+metadata:
+  name: gemma2-2b-cpu
+spec:
+  features: [TextGeneration]
+  url: ollama://gemma2:2b
+  engine: OLlama
+  resourceProfile: cpu:2
+```
+
+### Install Text Generation Model using L4 GPU
+
+Apply the following Model CRD to install the Llama 3.1 8B model using vLLM on L4 GPU:
+```yaml
+apiVersion: kubeai.org/v1
+kind: Model
+metadata:
+  name: llama-3.1-8b-instruct-fp8-l4
+spec:
+  features: [TextGeneration]
+  url: hf://neuralmagic/Meta-Llama-3.1-8B-Instruct-FP8
+  engine: VLLM
+  args:
+    - --max-model-len=16384
+    - --max-num-batched-token=16384
+    - --gpu-memory-utilization=0.9
+    - --disable-log-requests
+  resourceProfile: nvidia-gpu-l4:1
+```
+
+## Interact with the Text Generation Model
+The KubeAI service exposes an OpenAI compatible API that you can use to query the available models and interact with them.
+
+The KubeAI service is available at `http://kubeai/openai/v1` within the Kubernetes cluster.
+
+You can also port-forward the KubeAI service to your local machine to interact with the models:
+
+```bash
+kubectl port-forward svc/kubeai 8000:80
+```
+
+You can now query the available models using curl:
+
+```bash
+curl http://localhost:8000/openai/v1/models
+```
+
+### Using curl to interact with the model
+
+Run the following curl command to interact with the model named `llama-3.1-8b-instruct-fp8-l4`:
+```bash
+curl "http://localhost:8000/openai/v1/chat/completions" \
+    -H "Content-Type: application/json" \
+    -d '{
+        "model": "llama-3.1-8b-instruct-fp8-l4",
+        "messages": [
+            {
+                "role": "system",
+                "content": "You are a helpful assistant."
+            },
+            {
+                "role": "user",
+                "content": "Write a haiku about recursion in programming."
+            }
+        ]
+    }'
+```
+
+### Using the OpenAI Python SDK to interact with the model
+Once the pod is ready, you can use the OpenAI Python SDK to interact with the model:
+All OpenAI SDKs work with KubeAI since the KubeAI service is OpenAI API compatible.
+
+See the below example code to interact with the model using the OpenAI Python SDK:
```

Comment:
Alt: we could just link to different examples to keep docs easier to maintain. This would have the added benefit of directing users to the repo where they could add stars, issues, etc.

Reply:
I think we should embed code from the repo in docs in the future, but I do think from user pov that having it in the same doc is preferred.